src/main/java/ecdar/utility/helpers/ZoomHelper.java

```diff
@@ -41,74 +41,71 @@ public Double getZoomLevel() {
-        if (active && model != null) {
-            currentZoomFactor.set(zoomLevel);
-        }
+        if (!active || model == null) return;
+
+        currentZoomFactor.set(zoomLevel);
     }
 
     /**
      * Zoom in with a delta of 1.2
      */
     public void zoomIn() {
-        if (active) {
-            double delta = 1.2;
-            double newScale = currentZoomFactor.get() * delta;
+        if (!active) return;
 
-            //Limit for zooming in
-            if (newScale > maxZoomFactor) {
-                return;
-            }
+        double delta = 1.2;
+        double newScale = currentZoomFactor.get() * delta;
 
-            currentZoomFactor.set(newScale);
+        //Limit for zooming in
+        if (newScale > maxZoomFactor) {
+            return;
         }
+
+        currentZoomFactor.set(newScale);
     }
 
     /**
      * Zoom out with a delta of 1.2
      */
     public void zoomOut() {
-        if (active) {
-            double delta = 1.2;
-            double newScale = currentZoomFactor.get() / delta;
+        if (!active) return;
 
-            //Limit for zooming out
-            if (newScale < minZoomFactor) {
-                return;
-            }
+        double delta = 1.2;
+        double newScale = currentZoomFactor.get() / delta;
 
-            currentZoomFactor.set(newScale);
+        //Limit for zooming out
+        if (newScale < minZoomFactor) {
+            return;
         }
+
+        currentZoomFactor.set(newScale);
     }
 
     /**
      * Set the zoom multiplier to 1
      */
     public void resetZoom() {
         currentZoomFactor.set(1);
+        if (canvasPresentation
+                .getController()
+                .getActiveModelPresentation() instanceof DeclarationsPresentation) alignDeclaration();
     }
 
     /**
      * Zoom in to fit the component on screen
      */
     public void zoomToFit() {
-        if (active) {
-            if (!(canvasPresentation.getController().getActiveModelPresentation() instanceof ComponentPresentation || canvasPresentation.getController().getActiveModelPresentation() instanceof SystemPresentation)) {
-                resetZoom();
-                return;
-            }
-
-            double neededWidth = (model instanceof ComponentPresentation ?
-                    (model.getMinWidth()
-                            + ((ComponentController) model.getController()).inputSignatureContainer.getWidth()
-                            + ((ComponentController) model.getController()).outputSignatureContainer.getWidth())
-                    : model.getMinWidth());
+        if (!active || model == null) return;
 
+        double neededWidth = (model instanceof ComponentPresentation ?
+                (model.getMinWidth()
+                        + ((ComponentController) model.getController()).inputSignatureContainer.getWidth()
+                        + ((ComponentController) model.getController()).outputSignatureContainer.getWidth())
```

Comment:
Consider moving ```(ComponentController) model.getController()``` to a variable declaration. This could reduce complexity.

Reply:
I have introduced a variable and extracted this logic into a method, to hopefully improve readability

Reply:
As the Romulans might say: "Vah n'ghahk tr'khhae"